examples/north_sea/model_config.py

```diff
@@ -152,23 +152,33 @@ def construct_solver(spinup=False, store_station_time_series=True, **model_optio
             )
             solver_obj.add_callback(cb)
 
-    # Setup forcings
-    data_dir = os.path.join(os.environ.get("DATA", "./data"), "tpxo")
-    if not os.path.exists(data_dir):
-        raise IOError(f"Data directory {data_dir} does not exist")
-    forcing_constituents = ["Q1", "O1", "P1", "K1", "N2", "M2", "S2", "K2"]
     elev_tide_2d = Function(solver_obj.function_spaces.P1_2d, name="Tidal elevation")
-    tbnd = forcing.TPXOTidalBoundaryForcing(
-        elev_tide_2d,
-        start_date,
-        coord_system,
-        data_dir=data_dir,
-        constituents=forcing_constituents,
-        boundary_ids=[100],
-    )
-
-    # Set time to zero for the tidal forcings
-    tbnd.set_tidal_field(0.0)
+    if os.getenv("THETIS_REGRESSION_TEST") is None:
+        # Setup forcings
+        data_dir = os.path.join(os.environ.get("DATA", "./data"), "tpxo")
+        if not os.path.exists(data_dir):
+            raise IOError(f"Data directory {data_dir} does not exist")
+        forcing_constituents = ["Q1", "O1", "P1", "K1", "N2", "M2", "S2", "K2"]
+        tbnd = forcing.TPXOTidalBoundaryForcing(
+            elev_tide_2d,
+            start_date,
+            coord_system,
+            data_dir=data_dir,
+            constituents=forcing_constituents,
+            boundary_ids=[100],
+        )
+
+        # Set time to zero for the tidal forcings
+        tbnd.set_tidal_field(0.0)
+    else:
+        # for CI testing setup dummy tbnd
+        elev_tide_2d.assign(1.)
+
+        def set_tidal_field(t):
+            pass
+
+        tbnd = types.SimpleNamespace()
+        tbnd.set_tidal_field = set_tidal_field
```

Comment:
Nice, I haven't seen this approach before. Looks similar to using `unittest.mock`.